.github/workflows/java-windows.yml

```diff
@@ -17,12 +19,38 @@ jobs:
                 with:
                     java-version: ${{ matrix.java_version }}
 
+            -   name: Cache gradle dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ~/.gradle/caches
+                    key: ${{ runner.os }}-gradle-${{matrix.java_version}}-${{ hashFiles('**/*.gradle*') }}
```

Comment:
what is the reason that you want to cache the gradle dependencies per java version? It's always the same and thus could be shared no?

Reply:
Same as for android, since we differentiate builds per java we might want to differentiate these as well

Reply:
In thos case please remove as well